Source/Global/mem.h

```diff
@@ -75,6 +90,21 @@ class http_stl_allocator
     }
 };
 
+template<typename T>
+std::shared_ptr<T> http_allocate_shared()
```

Comment:
template<typename T, typename... Args>
std::shared_ptr<T> http_allocate_shared(Args&&... args)
{
    return std::allocate_shared<T>(http_stl_allocator<T>(), std::forward<Args>(args)...);
}

1. Made into a variadic template to allow passing arguments through to the constructor
2. Avoid specifying template arguments that can be deduced
3. Written in the browser so it may fail to compile

Reply:
done